src/entities/Customer.java

```diff
@@ -181,7 +189,7 @@ private String makeBooking(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB)
 				}
 			}
 			String TID = Integer.toString(cinema.getName().hashCode()%1000) + showtime.toString();
-			Ticket newTicket = new Ticket(mov, cinema, this, cinema.getSeat(title+showtime.toString(), index), TID, this.username, this.email, this.mobileNumber);
+			Ticket newTicket = new Ticket(mov, cinema, customer?? , cinema.getSeat(title+showtime.toString(), index), TID, this.username, this.email, this.mobileNumber);
```

Comment:
Ticket class constructors args and this differs. Need to decide on a fixed format